sdk/keyvault/keyvault-certificates/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -93,35 +128,37 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
       throw new Error("The resource address for authorization must use the 'https' protocol.");
     }
 
-    let originalBody = webResource.body;
+    const originalBody = webResource.body;
 
-    if (this.challenge == undefined) {
+    if (this.challengeCache.challenge == undefined) {
       // Use a blank to start the challenge
       webResource.body = "";
     } else {
       // or use the cached token if we have one
       await this.authenticateRequest(webResource);
     }
 
-    let response = await this._nextPolicy.sendRequest(webResource);
+    const response = await this._nextPolicy.sendRequest(webResource);
 
     if (response.status == 401) {
       webResource.body = originalBody;
 
       let www_authenticate = response.headers.get("WWW-Authenticate");
 
       if (www_authenticate) {
-        let resource = this.parseWWWAuthenticate(www_authenticate);
-        let challenge = new AuthenticationChallenge(resource + "/.default")
+        const { authorization, resource } = this.parseWWWAuthenticate(www_authenticate);
```

Comment:
can we document what `authorization` is here?  Is it the cache, some kind of token in the cache, etc?

Reply:
What do you think of a comment like:

```
// The challenge based authentication will contain both an authorization URI with a token,
// and the resource to which that token is valid against (also called the scope).
```

Reply:
Pushed the idea on the previous comment here: https://github.com/Azure/azure-sdk-for-js/pull/8558/commits/6e8863ff3d8493b68e74640c242ee0a93566a321